src/libraries/System.Text.Json/docs/ParameterizedCtorSpec.md

```diff
@@ -0,0 +1,862 @@
+# Deserializing objects using parameterized constructors with `JsonSerializer`
+
+## Motivation
+
+`JsonSerializer` deserializes instances of objects (`class`es and `struct`s) using public parameterless
+constructors. If none is present, and deserialization is attempted, the serializer throws a `NotSupportedException`
+with a message stating that objects without public parameterless constructors, including `interface`s and `abstract`
+types, are not supported for deserialization. There is no way to deserialize an instance of an object using a parameterized constructor.
+
+A common pattern is to make data objects immutable for various reasons. For example, given `Point`:
```

Comment:
Please note that this change alone won't support all immutable types. For example, our own immutable collections don't use constructors but factory methods.